source/src_lcao/gint_gamma_vl.cpp

```diff
@@ -377,15 +377,8 @@ Gint_Tools::Array_Pool<double> Gint_Gamma::gamma_vlocal(const double*const vloca
 						//------------------------------------------------------
 						// index of wave functions for each block
 						//------------------------------------------------------
-						int *block_iw = Gint_Tools::get_block_iw(na_grid, grid_index, this->max_size);
-						
-						int* block_index = Gint_Tools::get_block_index(na_grid, grid_index);
-						
-						//------------------------------------------------------
-						// band size: number of columns of a band
-						//------------------------------------------------------
-						int* block_size = Gint_Tools::get_block_size(na_grid, grid_index);
-
+                        int * block_iw, * block_index, * block_size;
```

Comment:
In this refactor, you used raw pointer and allocated it in another function, why not use "vector<int>" and "resize" to avoid to manage the memory.

Reply:
I was following the old practice, will change to vector

Reply:
> In this refactor, you used raw pointer and allocated it in another function, why not use "vector" and "resize" to avoid to manage the memory.

Changing it to vector will cause some strange errors 'Intel MKL ERROR: Parameter 8 was incorrect on entry to DGEMM' so it stays this way for now.

Reply:
> Changing it to vector will cause some strange errors 'Intel MKL ERROR: Parameter 8 was incorrect on entry to DGEMM' so it stays this way for now.
Hi @wenfei-li ,
you can call `DGEMM` with the `data()` method of a vector to get its actual address.